### DISCUSSION TITLE
Haveno - Bisq fork from the Monero community

### DISCUSSION THREAD

**Discussion by chimp1984** [bisq-network/bisq#5429] [General]
I looked a bit into the [haveno project](https://github.com/haveno-dex/haveno) and wanted to share my impression/opinion.

They have forked the Bisq code a while back to evaluate a proof of concept to use XMR as base currency and it seems they got the basic funtionality working using Woodser's java rpc library for the Monero functionality.
They want to use a 2of3 multisig with an arbitrator who will be one of the team members. Monero has very limited smart contract capabilities. There is no timelock and you cannot created chained transactions. Funds from a transation can only be spent after 10 blocks/20 min. All that restricts the potential options a lot. The current protocol Bisq uses would requires a time locked tx so that cannot be implemented (beside that they don't want to use the DAO anyway). The maker fee tx and taker fee tx play several roles which become problematic in their context.
They currently plan to keep those transactions but that comes with usability costs (time) and higher failure rate of trades due the time delay.
A maker need to wait 20 min after creating an offer until it can be taken. That might be not a major issue if the publishing is just delayed or its publsihed immediately but cannot be taken before 10 confirmations. 
But when a taker takes the offer the taker fee tx also requires 10 blocks. Here its really problematic as the maker might go offline in those 20 min and then the taker has lost his trade fee and all ends up in a failed trade.
After that the deposit tx requires another 20 min (compared to 10 min in Bisq), so thats not a big problem but also not an improvement. And then after the trade is completed the payout tx will require again the 20 min before the user could use those funds for another trade. Doing multiple offers and trades in a row will be a time consuming task. 

I think they have to remove the taker fee tx at least, which does not cause too much troubles. Removing the maker fee tx would be more difficult as that serves to keep the utxo for the trade resevered and not having that requires more complex wallet management. Also not having a maker fee tx adds vulnerability for offer spam. 

My biggest concern though is their decision to go back to the trusted 3rd party model of a 2of3 multisig. That comes with legal and security risks and privacy and scalability issues.  
But I also do not see any other solution as the current Bisq model using the DAO instead cannot work with Monero even if they would keep the DAO.

I was considering myself to use XMR as base layer with a 2of2 Multisig MAD (mutual assured destruction) model, but came to the conclusion that this is too risky and would not find acceptance. What I have learned recently about the limitations of Monero reduced my hope that there is a viable solution using Monero as base layer. A pity as the strong privacy and low miner fees would have made it a perfect candidate.

**Comment by chimp1984** [comment]
Here a little bit of an update after engaging in some discussions on their Matrix channel with some Haveno team members.
I tried to bring up some of my concerns and also suggested some solutions (like removing the maker and taker fee tx), but my main concern regarding the risk of introducing a trusted 3rd party got heavily ignored/rejected. They are convinced that regulators have signed off that partizipating in a multisig is not subject to regulation. I guess if that statement is true it mainly refers to the use case where the 2rd party is used for security reasons but never can spend the funds without the signature of the owner, but not the 2of3 use case where 2 owners are at play with an additional party which could collude with one side and steal the funds. 
But anyway, however the real legal situation is, the overall reaction to ignore my valid concerns and criticism and defamate me and Bisq repeatedly is telling about the quality of that team (I exclude here the main dev as it seems he is a very solid dev, as well as Justin who was more open to some discussion).

We had in the past issues with that spreading of FUD at Reddit after the failed Monero wallet integration project  and erciccione a main team member at the Haveno project earned quite a lot of negative reputation by repeating FUD about Bisq (that Bisq does not care about the Monero community) even we communicated very clearly our reasons why we did not integrate the wallet (security concerns).    

So my conclusion is that a team with such a mindset will not succeed to solve a very hard problem in a very hostile regulatory environment. It is a bit frustrating to see how low the quality of communication is sometimes in the crypto space and I am aware that there are lot of excellent people in the Monero community, but what I observed in the Haveno community, does not reflect those.

**Reply by m52go** [reply]
Similar situation [on Twitter](https://twitter.com/bisq_network/status/1385657728736006146) yesterday. So much negative energy. I think they have their heads in the clouds from all the excitement but will be humbled quickly.

**Reply by chimp1984** [reply]
While copycats and trolls move back to past models we move forward: https://github.com/bisq-network/proposals/issues/330 Best to just ignore those trolls.

**Comment by erciccione** [comment]
@chimp1984 I ignored your comments initially, since they appear to be crafted to sparkle drama, but the amount of falsities being spread is reaching a level that cannot be ignored.

> the overall reaction to ignore my valid concerns and criticism and defamate me and Bisq repeatedly

This is is provably false. When you came in the #haveno chat with a different nick to share your criticisms, we happily received it and i personally thanked you and assured you your concerns were noted, most of which we were already aware of and we are discussing internally. The problems begun when days later you started to PM members of the Haveno team privately, repeating the same criticisms (even if you already got an answer from the same people in public) but adding a more apocalyptic tone and small, incorrect, details that would picture a much more dramatic situation than it actually is. That wasn't well received by me, as i received it like an attempt to scare key Haveno contributors and spread FUD about the whole project, but i and anybody else never insulted or trolled you. The chat logs are public and anyone joining `#haveno` or `#haveno-dev` on matrix can read the backlog.

> I guess if that statement is true

I strongly suggest you to stop making guesses. Most of what's written in your OP about how Haveno will work are assumptions, based on nothing but your perception of the state of things. We haven't finalized our trade protocol, so your guesses on how the UX will be are just that, guesses. You are obviously free to guess whatever you want, but please don't make these assumptions public as if they were universal truths.

Also, wile you guess how the 2/3 multisig is regulated, we are in direct contact with these regulators, and we are studying and following their guidelines with the help of experts in the field.

&nbsp;

Anyway, i'm not interested in feeding drama, but at least i'm happy to see that as a result of Haveno being under development, Bisq received the wake up call and it's speeding up their protocol improvements to offer a better UX.

Your concerns are noted, but honestly sadden me to see your reactions. It's not unexpected though. If Haveno is successful bisq could lose 90% or more of its revenue and as you know, the vast majority of Bisq's revenue come from Monero trades and our main goal is to prioritize those, so i understand you feel threatened.

@m52go

> So much negative energy

As pointed out multiple times, Monero people have been asking Bisq to improve their XMR support for years, but without any substantial result (and no, adding automatic transaction confirmations is not a substantial enough improvement IMHO). The failed integrations and the unwillingness of Bisq's team to prioritize Monero have frustrated the XMR community, who desperately need a decentralized XMR <-> fiat exchange. Bisq hasn't delivered (for whatever reason, it doesn't matter) so we decided to assemble a team and take care of the problem ourselves. This shouldn't be seen as an attack or as trolling, but as healthy competition.

The road is long and hard, but we believe in the project and we are working to make it real. Any help and suggestions are appreciated and we welcome every Bisq contributor willing to contribute to the Haveno project or to share constructive criticism. Good luck :)

**Reply by m52go** [reply]
Being disappointed with the decisions of a project, forking it, and providing healthy competition is one thing. Attacking and trolling is another. You and some of the Monero community have done both. Please keep the latter away from this project.

**Comment by erciccione** [comment]
Hello everybody.

Just a heads up for people stumbling on this discussion now or in the future. We published our final trade protocol, which is available in detail here: https://github.com/haveno-dex/haveno/tree/master/docs/trade_protocol (3 documents available, the quick sketch of the protocol is in `trade-protocol.md `)

We are also in the process of implementing it: https://github.com/haveno-dex/haveno/pull/107

If you have questions, we will be very happy to answer ([our FAQ could be useful as well](https://github.com/haveno-dex/haveno/wiki/FAQ)) here or in our public channels (`#haveno` and `#haveno-dev` on Matrix and IRC).

**Comment by bwusth** [comment]
Has anyone suggested or attempted a fork of Bisq which would use Zcash as the base traded currency?
